frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -0,0 +1,675 @@
+        fn on_initialize(_now: BlockNumberFor<T>) -> Weight {
+            // just return the weight of the on_finalize.
+            T::DbWeight::get().reads(1)
+        }
+
+        fn on_finalize(_n: BlockNumberFor<T>) {
+            // TODO: era calculation
+            // Set the start of the first era.
+            if let Some(mut active_era) = Self::active_era() {
+                if active_era.start.is_none() {
+                    let now_as_millis_u64 = T::UnixTime::now().as_millis().saturated_into::<u64>();
+                    active_era.start = Some(now_as_millis_u64);
+                    // This write only ever happens once, we don't include it in the weight in general
+                    ActiveEra::<T>::put(active_era);
+                }
+            }
+            // `on_finalize` weight is tracked in `on_initialize`
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Take the origin account as a stash and lock up `value` of its balance. `controller` will
+        /// be the account that controls it.
+        ///
+        /// `value` must be more than the `minimum_balance` specified by `T::Currency`.
+        ///
+        /// The dispatch origin for this call must be _Signed_ by the stash account.
+        ///
+        /// # <weight>
+        /// - Independent of the arguments. Moderate complexity.
+        /// - O(1).
+        /// - Three extra DB entries.
+        ///
+        /// NOTE: Two of the storage writes (`Self::bonded`, `Self::payee`) are _never_ cleaned unless
+        /// the `origin` falls below _existential deposit_ and gets removed as dust.
+        /// # </weight>
+        #[pallet::weight(T::WeightInfo::bond())]
+        pub fn bond(
+            origin: OriginFor<T>,
+            controller: <T::Lookup as StaticLookup>::Source,
+            #[pallet::compact] value: BalanceOf<T>,
+            payee: RewardDestination<T::AccountId>,
+        ) -> DispatchResultWithPostInfo {
+            let stash = ensure_signed(origin)?;
+            ensure!(
+                !Bonded::<T>::contains_key(&stash),
+                Error::<T>::AlreadyBonded
+            );
+
+            let controller = T::Lookup::lookup(controller)?;
+            ensure!(
+                !Ledger::<T>::contains_key(&controller),
+                Error::<T>::AlreadyPaired
+            );
+
+            // reject a bond which is considered to be dust
+            ensure!(
+                value >= T::Currency::minimum_balance(),
+                Error::<T>::InsufficientValue
+            );
+
+            let stash_balance = T::Currency::free_balance(&stash);
+            ensure!(!stash_balance.is_zero(), Error::<T>::InsufficientValue);
+
+            Bonded::<T>::insert(&stash, &controller);
+            Payee::<T>::insert(&stash, payee);
+
+            Self::deposit_event(Event::Bonded(stash.clone(), stash_balance.clone()));
+
+            let value = value.min(stash_balance);
```

Comment:
@Maar-io Probably a silly question - is this by design? E.g. if user wants to bond 1000 but only has 100 available, no error will be raised, instead just 100 will be bonded?

I can also see the check on line 262 which just requires user to have something free.

Reply:
I expect it to be same as with bonding on the node staking. The transaction will not be possible since you do not have enough to pay the fee.
![image](https://user-images.githubusercontent.com/34627453/131973466-7f6fa6ef-2d38-4e76-949e-75538cf90060.png)